Mathlib/MeasureTheory/Function/LpSeminorm/Basic.lean

```diff
@@ -1102,6 +1102,21 @@ theorem eLpNorm'_le_nnreal_smul_eLpNorm'_of_ae_le_mul {f : α → F} {g : α →
   simp_rw [ENNReal.coe_le_coe, ← NNReal.mul_rpow, NNReal.rpow_le_rpow_iff hp]
   exact h
 
+-- XXX: should the nnnorm version be deleted or primed instead?
+theorem eLpNorm'_le_nnreal_smul_eLpNorm'_of_ae_le_mul' {f : α → ε} {g : α → ε'} {c : ℝ≥0}
+    (h : ∀ᵐ x ∂μ, ‖f x‖ₑ ≤ c * ‖g x‖ₑ) {p : ℝ} (hp : 0 < p) :
+    eLpNorm' f p μ ≤ c • eLpNorm' g p μ := by
```

Comment:
Isn't this true?
```suggestion
theorem eLpNorm'_le_mul_eLpNorm'_of_ae_le_mul' {f : α → ε} {g : α → ε'} {c : ℝ≥0∞}
    (h : ∀ᵐ x ∂μ, ‖f x‖ₑ ≤ c * ‖g x‖ₑ) {p : ℝ} (hp : 0 < p) :
    eLpNorm' f p μ ≤ c * eLpNorm' g p μ := by
```

Reply:
Good question! Kind of, but not quite.
- if `c ≠ ⊤`, this is true and basically the same proof works
- if `c = ⊤` but `eLpNorm' g ≠ 0`, this is also true: in that case, the lemma is almost vacuous
- the case `c = ⊤` and `eLpNorm' g = 0` is more interesting: recall that `0 * \top = 0`, so we need to prove something.
If* `g` is measurable, then g = 0 a.e. and hypothesis `h` implies `f = 0` a.e., so the statement holds.
The issue is if `‖g ·‖ₑ ^ p` is not measurable, as then its integral (which is `eLpNorm' g`) is always zero. In that case, hypothesis `h` does not imply that `f = 0` a.e., so we have a problem.

(Lean is great for detecting these edge cases.)

Reply:
To summarise: I can easily
- add a version with multiplication
- also allow c = \top --- with *some* additional hypothesis.
What do you think?

Reply:
A version with multiplication assuming measurability would be great, yes!

Reply:
Added - modulo one small sorry where I'm not sure yet how to do it in Lean.